pkg/util/vmopv1/vm.go

```diff
@@ -138,3 +141,71 @@ func ResolveImageName(
 
 	return obj, nil
 }
+
+// DetermineHardwareVersion returns the hardware version recommended for the
+// provided VirtualMachine based on its own spec.minHardwareVersion, as well as
+// the hardware in the provided ConfigSpec and requirements of the given
+// VirtualMachineImage.
+func DetermineHardwareVersion(
+	vm vmopv1.VirtualMachine,
+	configSpec vimtypes.VirtualMachineConfigSpec,
+	imgStatus vmopv1.VirtualMachineImageStatus) vimtypes.HardwareVersion {
+
+	// Get the minimum hardware version required by the VM from the VM spec.
+	vmMinVersion := vimtypes.HardwareVersion(vm.Spec.MinHardwareVersion)
+
+	// Check to see if the ConfigSpec specifies a hardware version.
+	var configSpecVersion vimtypes.HardwareVersion
+	if configSpec.Version != "" {
+		configSpecVersion, _ = vimtypes.ParseHardwareVersion(configSpec.Version)
+	}
+
+	// If the ConfigSpec contained a valid hardware version, then the version
+	// the VM will use is determined by comparing the ConfigSpec version and the
+	// one from the VM's spec.minHardwareVersion field and returning the largest
+	// of the two versions.
+	if configSpecVersion.IsValid() {
+		return max(vmMinVersion, configSpecVersion)
+	}
+
+	// A VM Class with an embedded ConfigSpec should have the version set, so
```

Comment:
would probably make this more generic for future devs without the context of the embedded configSpec and a configSpec from HW devices
	// A VM Class with an embedded ConfigSpec should have the version set, so
	// this is a ConfigSpec we created from the HW devices in the class.

Reply:
```suggestion
	// A VM Class with an embedded ConfigSpec will in most cases have a version set, for brownfield 
	// VM Classes where the configSpec was synthesized from the HW devices .. blah blah.. 
```
maybe?

Reply:
It was an existing comment, but I can update it, sure. Thanks!